pkg/meta/signatures/signatures.go

```diff
@@ -21,18 +29,106 @@ const (
 	defaultFilePerms  = 0o644
 )
 
-func InitCosignAndNotationDirs(rootDir string) error {
-	err := InitCosignDir(rootDir)
+type SigStore struct {
+	CosignStorage   publicKeyStorage
+	NotationStorage certificateStorage
+}
+
+func NewLocalSigStore(rootDir string) (*SigStore, error) {
+	publicKeyStorage, err := NewPublicKeyLocalStorage(rootDir)
+	if err != nil {
+		return nil, err
+	}
+
+	certStorage, err := NewCertificateLocalStorage(rootDir)
+	if err != nil {
+		return nil, err
+	}
+
+	return &SigStore{
+		CosignStorage:   publicKeyStorage,
+		NotationStorage: certStorage,
+	}, nil
+}
+
+func NewCloudSigStore(region, endpoint string) (*SigStore, error) {
+	secretsManagerClient, err := GetSecretsManagerClient(region, endpoint)
+	if err != nil {
+		return nil, err
+	}
+
+	secretsManagerCache, err := GetSecretsManagerRetrieval(region, endpoint)
+	if err != nil {
+		return nil, err
+	}
+
+	publicKeyStorage := NewPublicKeyCloudStorage(secretsManagerClient, secretsManagerCache)
+
+	certStorage, err := NewCertificateCloudStorage(secretsManagerClient, secretsManagerCache)
+	if err != nil {
+		return nil, err
+	}
+
+	return &SigStore{
+		CosignStorage:   publicKeyStorage,
+		NotationStorage: certStorage,
+	}, nil
+}
+
+func GetSecretsManagerClient(region, endpoint string) (*secretsmanager.Client, error) {
+	customResolver := aws.EndpointResolverWithOptionsFunc(
```

Comment:
we already have this logic duplicated for dynamodb client, you can split both GetDynamoClient and GetSecretsManagerClient in two parts,
func GetAwsConfig(region, endpoint) aws.Config {}
+
func GetDynamoClient(awsConfig) {}
func GetSecretsManagerClient(awsConfig) {}